app/src/org/commcare/gis/MapboxLocationPickerViewModel.kt

```diff
@@ -0,0 +1,68 @@
+package org.commcare.gis
+
+import android.app.Application
+import android.location.Geocoder
+import android.location.Location
+import androidx.lifecycle.AndroidViewModel
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.viewModelScope
+import com.mapbox.mapboxsdk.geometry.LatLng
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+import org.commcare.dalvik.R
+import org.commcare.utils.GeoUtils
+import org.commcare.utils.StringUtils
+import java.util.*
+
+/**
+ * @author $|-|!˅@M
+ */
+class MapboxLocationPickerViewModel(application: Application): AndroidViewModel(application) {
+
+    val placeName = MutableLiveData<String>()
+    private var location = Location("XForm")
+
+    fun reverseGeocode(latLng: LatLng) {
+        viewModelScope.launch(Dispatchers.IO) {
+            var addressString = ""
+            val geocoder = Geocoder(getApplication(), Locale.getDefault())
+            val geoAddress = geocoder.getFromLocation(latLng.latitude, latLng.longitude, 1)
```

Comment:
Thanks for pointing this! `getFromLocation` throws an `IOException` when there's no network which I forgot to check earlier. I thought it'd simply return null and I handled that case already by showing the coordinates instead of address.